examples/common/entry/entry_sdl.cpp

```diff
@@ -323,9 +323,7 @@ namespace entry
 		SDL_UserEvent& uev = event.user;
 		uev.type = s_userEventStart + _type;
 
-		union { void* p; WindowHandle h; } cast;
-		cast.h = _handle;
-		uev.data1 = cast.p;
+		uev.data1 = reinterpret_cast<void*>(_handle);
```

Comment:
One benefit of union here is that type-safety will actually error if handle doesn't match. Where reinterpret_cast will just do whatever.

Reply:
You bring up an excellent point about the importance of type checking. It's crucial to ensure stability, especially in scenarios where `_handle` might change its type unexpectedly. For instance, if the API were to transition `WindowHandle` to an integer type like `uint64_t`, it could lead to complications if the cast to `void*` isn't adjusted accordingly.

I think we can enhance our type checking to be more robust and explicit by incorporating `bx::isSame<Ty, Uy>()`. This approach emphasizes clarity and intentionality, making it easier to understand the expected behavior. I'll demonstrate this further with an example in an upcoming commit.